src/domain/position/binProcessor.js

```diff
@@ -0,0 +1,242 @@
+ --------------
+ ******/
+'use strict'
+
+const Logger = require('@mojaloop/central-services-logger')
+const BatchPositionModel = require('../../models/position/batch')
+const BatchPositionModelCached = require('../../models/position/batchCached')
+const PositionPrepareDomain = require('./prepare')
+const SettlementModelCached = require('../../models/settlement/settlementModelCached')
+const Enum = require('@mojaloop/central-services-shared').Enum
+const ErrorHandler = require('@mojaloop/central-services-error-handling')
+// TODO: We may not need this if we optimize the participantLimit query
+const participantFacade = require('../../models/participant/facade')
+
+/**
+ * @function processBins
+ *
+ * @async
+ * @description This is the domain function to process a list bins containing position messages grouped by participant account.
+ *
+ * @param {array} bins - a list of account-bins to process
+ * @param {object} trx - Database transaction object
+ *
+ * @returns {results} - Returns a list of bins with results or throws an error if failed
+ */
+const processBins = async (bins, trx) => {
+  const transferIdList = []
+  await iterateThroughBins(bins, async (_accountID, _action, item) => {
+    if (item.decodedPayload?.transferId) {
+      transferIdList.push(item.decodedPayload.transferId)
+    }
+  })
+  // Pre fetch latest transferStates for all the transferIds in the account-bin
+  const latestTransferStateChanges = await BatchPositionModel.getLatestTransferStateChangesByTransferIdList(trx, transferIdList)
+  const latestTransferStates = {}
+  for (const key in latestTransferStateChanges) {
+    latestTransferStates[key] = latestTransferStateChanges[key].transferStateId
+  }
+
+  const accountIds = Object.keys(bins)
+
+  // Pre fetch all settlement accounts corresponding to the position accounts
+  // Get all participantIdMap for the accountIds
+  const participantCurrencyIds = await BatchPositionModelCached.getParticipantCurrencyByIds(trx, accountIds)
+
+  // TODO: Validate if all the participantCurrencyIds exist for all the accountIds
+
+  const allSettlementModels = await SettlementModelCached.getAll()
+
+  // Construct an object participantIdMap, accountIdMap amd currencyIdMap
+  const participantIdMap = {}
+  const accountIdMap = {}
+  const currencyIdMap = {}
+  for (const item of participantCurrencyIds) {
+    const { participantId, currencyId, participantCurrencyId } = item
+    if (!participantIdMap[participantId]) {
+      participantIdMap[participantId] = {}
+    }
+    if (!currencyIdMap[currencyId]) {
+      currencyIdMap[currencyId] = {
+        settlementModel: _getSettlementModelForCurrency(currencyId, allSettlementModels)
+      }
+    }
+    participantIdMap[participantId][currencyId] = participantCurrencyId
+    accountIdMap[participantCurrencyId] = { participantId, currencyId }
+  }
+
+  // TODO: Verify if all the accountIds have a corresponding participantCurrencyId
+  // TODO: Verify all maps are correctly constructed
```

Comment:
More than two TODOs in this file. If we ran sonarqube check on this it would put us in the red just for this alone without any further criteria..

Is there a way to address this?